smart-contract-verifier/src/vyper/compiler.rs

```diff
@@ -0,0 +1,190 @@
+use std::path::Path;
+
+use ethers_solc::{error::SolcError, CompilerInput, CompilerOutput, Solc};
+
+use crate::compiler::{self, EvmCompilerAgent};
+
+pub struct VyperCompilerAgent {}
+
+impl VyperCompilerAgent {
+    pub fn new() -> Self {
+        VyperCompilerAgent {}
+    }
+}
+
+impl EvmCompilerAgent for VyperCompilerAgent {
+    fn compile(
+        &self,
+        path: &Path,
+        ver: &compiler::Version,
+        input: &CompilerInput,
+    ) -> Result<CompilerOutput, SolcError> {
+        let _ = ver;
+        let vyper_output: types::VyperCompilerOutput = Solc::from(path).compile_as(input)?;
+        Ok(CompilerOutput::from(vyper_output))
+    }
+}
+
+mod types {
+    use std::collections::BTreeMap;
+
+    use ethers_solc::{
+        artifacts::{Contract, Severity},
+        CompilerOutput,
+    };
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+    pub struct SourceLocation {
+        pub file: String,
+    }
+
+    #[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+    #[serde(rename_all = "camelCase")]
+    pub struct VyperError {
```

Comment:
We may name it just an `Error` as we are working on in order not to repeat `vyper` module prefix
```suggestion
    pub struct Error {
```

Reply:
we won't repear vyper module prefix, since this module is private

if we rename this, we have to rename VyperCompilerOutput also, then we will have a lot of name collisions, and write all solidity errors and outputs like `ethers_solc::artifacts::Error`. to be homogeneous we will have to write this prefix for ethers_solc::artifacts::Severity, ethers_solc::artifacts::Contract and so on.

I tried to write this way, it's just an unreadable mess